monkestation/code/modules/spells/spell_types/blinkdagger.dm

```diff
@@ -0,0 +1,42 @@
+/obj/effect/proc_holder/spell/targeted/blinkdagger
+	name = "Blink Dagger"
+	desc = "Teleports behind you"
+	invocation = "Nothing personnel, kid."
+	action_icon = 'icons/mob/actions/actions_spells.dmi'
+	action_icon_state = "blink"
+	charge_max = 30 SECONDS
+	invocation_type = "shout"
+	range = 7
+	selection_type = "range"
+	clothes_req = FALSE
+	invocation_time = 10
+
+/obj/effect/proc_holder/spell/targeted/blinkdagger/cast(list/targets, mob/user = usr)
+	if(!length(targets))
+		to_chat(user, "<span class='notice'>No target found in range.</span>")
+		return
+
+	var/mob/living/carbon/target = targets[1]
+
+	if(!(target in oview(range)))
+		to_chat(user, "<span class='notice'>[target.p_theyre(TRUE)] too far away!</span>")
+		revert_cast()
+		return
+
+	user.put_in_hands(new /obj/item/melee/blinkknife(src))
```

Comment:
This should probably just be the same knife every time.
Put a define variable (and destroy call) for it so you aren't just creating extra daggers.
As it stands, using this ability three times will leave a dagger on the floor and two in your hands.

Reply:
5 second qdel should fix this

Reply:
also I did some testing and turns out this is not a problem anyway